packages/nimble-components/src/table-column/enum-text/index.ts

```diff
@@ -0,0 +1,78 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { TableColumnEnumBase, TableColumnEnumColumnConfig } from '../enum-base';
+import { styles } from '../enum-base/styles';
+import { template } from '../enum-base/template';
+import { TableColumnSortOperation, TableColumnValidity } from '../base/types';
+import { mixinGroupableColumnAPI } from '../mixins/groupable-column';
+import { mixinFractionalWidthColumnAPI } from '../mixins/fractional-width-column';
+import { MappingText } from '../../mapping/text';
+import { TableColumnEnumTextValidator } from './models/table-column-enum-text-validator';
+import type { ColumnInternalsOptions } from '../base/models/column-internals';
+import { tableColumnEnumTextCellViewTag } from './cell-view';
+import { tableColumnEnumTextGroupHeaderViewTag } from './group-header-view';
+import type { Mapping } from '../../mapping/base';
+import type { MappingConfig } from '../enum-base/models/mapping-config';
+import { MappingTextConfig } from '../enum-base/models/mapping-text-config';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-enum-text': TableColumnEnumText;
+    }
+}
+
+/**
+ * Table column that maps values to strings
+ */
+export class TableColumnEnumText extends mixinGroupableColumnAPI(
+    mixinFractionalWidthColumnAPI(
+        TableColumnEnumBase<
+        TableColumnEnumColumnConfig,
+        TableColumnEnumTextValidator
+        >
+    )
+) {
+    public override createValidator(): TableColumnEnumTextValidator {
+        return new TableColumnEnumTextValidator(this.columnInternals, [
+            MappingText
+        ]);
+    }
+
+    public override get validity(): TableColumnValidity {
+        return this.validator.getValidity();
+    }
+
+    protected override getColumnInternalsOptions(): ColumnInternalsOptions {
+        return {
+            cellRecordFieldNames: ['value'],
+            cellViewTag: tableColumnEnumTextCellViewTag,
+            groupHeaderViewTag: tableColumnEnumTextGroupHeaderViewTag,
+            delegatedEvents: [],
+            sortOperation: TableColumnSortOperation.basic
+        };
+    }
+
+    protected override updateColumnConfig(): void {
+        this.validator.validate(this.mappings, this.keyType);
+        this.columnInternals.columnConfig = this.createColumnConfig();
+    }
```

Comment:
I think at this point this is completely generic (and the other branch shows identical implementations for each column type). Can we move this to the base class?

Even more "correctly" it is validation that is column specific. So the validator execution could be overridable but the `updateColumnConfig` algorithm is always the same, i.e. run validation then create the columnConfig.

Reply:
I had to make a few other changes to make this work properly. Now `createColumnConfig` is abstract and returns a generic type instead of `TableColumnEnumColumnConfig`. This way, classes derrived from `TableColumnEnumBase` can _actually_ have different column config types. Otherwise, we would get assignment errors when trying to do `this.columnInternals.columnConfig = this.createColumnConfig();`.